django/db/models/sql/compiler.py

```diff
@@ -330,8 +330,21 @@ def _order_by_pairs(self):
                 if not self.query.standard_ordering:
                     field = field.copy()
                     field.reverse_ordering()
+                src = field.get_source_expressions()[0]
+                if isinstance(src, F):
+                    try:
+                        resolved = src.resolve_expression(self.query, summarize=True)
```

Comment:
Do we really need to resolve this at this level? Or is there a way to mark it as reference without resolving?

Reply:
This seems to be on the right path but I'm not sure `query.annotations` is correct as some might have been excluded through `.values`? Do we want to check `.select` and `.extra_select` as well?

There must be a way to merge this logic with the one below that already cover all of these paths.